packages/vector_graphics_codec/lib/vector_graphics_codec.dart

```diff
@@ -530,15 +539,81 @@ class VectorGraphicsCodec {
     buffer._currentPathId = -1;
   }
 
+  /// Saves a copy of the current transform and clip on the save stack, and then
+  /// creates a new group which subsequent calls will become a part of. When the
+  /// save stack is later popped, the group will be flattened into a layer and
+  /// have the given `paint`'s [Paint.colorFilter] and [Paint.blendMode]
+  /// applied.
+  ///
+  /// See also:
+  ///   * [Canvas.saveLayer]
   void writeSaveLayer(VectorGraphicsBuffer buffer, int paint) {
     buffer._putUint8(_saveLayerTag);
     buffer._putUint16(paint);
   }
 
+  /// Pops the current save stack, if there is anything to pop.
+  /// Otherwise, does nothing.
+  ///
+  /// See also:
+  ///   * [Canvas.restore]
   void writeRestoreLayer(VectorGraphicsBuffer buffer) {
     buffer._putUint8(_restoreTag);
   }
 
+  /// Write the [text] contents given starting at [x], [y].
+  ///
+  /// [x], [y] are the coordinates of the starting point of the text baseline.
+  int writeTextConfig({
+    required VectorGraphicsBuffer buffer,
+    required String text,
+    required String? fontFamily,
+    required double dx,
+    required double dy,
+    required int fontWeight,
+    required double fontSize,
+  }) {
+    if (buffer._decodePhase.index > _CurrentSection.text.index) {
+      throw StateError('Text config must be encoded together.');
+    }
+    buffer._decodePhase = _CurrentSection.text;
+    final int textId = buffer._nextTextId++;
+    assert(textId < kMaxId);
+
+    buffer._putUint8(_textTag);
+    buffer._putUint16(textId);
+    buffer._putFloat32(dx);
+    buffer._putFloat32(dy);
+    buffer._putFloat32(fontSize);
+    buffer._putUint8(fontWeight);
+
+    // font-family
+    if (fontFamily != null) {
+      final Uint8List encoded = utf8.encode(fontFamily) as Uint8List;
+      buffer._putUint16(encoded.length);
+      buffer._putUint8List(encoded);
+    } else {
+      buffer._putUint16(0);
+    }
+
+    // text-value
+    final Uint8List encoded = utf8.encode(text) as Uint8List;
+    buffer._putUint16(encoded.length);
+    buffer._putUint8List(encoded);
```

Comment:
We should probably assert/throw if the text is empty or if the UTF8 encoding of it fails.

I also can't remember what the default is anymore, but I think we'll want to tolerate malformed UTF8.

Reply:
its not the default to tolerate it, but we can easily add that to the encoding. I can add a test case for that

Reply:
actually I don't see how this is possible. We parse this from a dart string which has to be valid utf8 or utf16

Reply:
Ahh ok, fair enough.